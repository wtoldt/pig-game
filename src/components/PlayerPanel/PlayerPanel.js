import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './PlayerPanel.scss';

export default class PlayerPanel extends Component {

  static propTypes = {
    player: PropTypes.object.isRequired,
    active: PropTypes.bool,
    left: PropTypes.bool,
    right: PropTypes.bool
  };

  static defaultProps = {
    active: false,
    left: false,
    right: false
  };

  render() {
    const { player: { name, overallScore, currentScore }, active, left, right } = this.props;
    const classnames = classNames('pg-player-panel', { active, left, right });
    return (
      <div className={classnames}>
        <div className="player-name-container">
          <div className="player-name">{ name }</div>
          <div className="overall-score">{ overallScore }</div>
        </div>
        <div className="current-score-container">
          <div className="current-score">{ currentScore }</div>
        </div>
      </div>
    );
  }
}