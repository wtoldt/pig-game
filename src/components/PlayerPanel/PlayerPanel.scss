@import '../../styles/consts.scss';

.pg-player-panel {
  flex: 1 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  background-color: white;

  &.active {
    background-color: lavenderblush;
    .player-name-container .player-name {
      color: rosybrown;
    }
  }

  &.left {
    border-top-left-radius: $big-panel-br;
    border-bottom-left-radius: $big-panel-br;
  }

  &.right {
    border-top-right-radius: $big-panel-br;
    border-bottom-right-radius: $big-panel-br;
  }

  .player-name, .overall-score, .current-score {
    width: 75px;
  }

  .player-name-container {
    .player-name {
      font-size: 60px;
      font-weight: 800;
      text-align: center;
      background-color: black;
      color: gray;
      padding-top: 30px;
    }
    .overall-score {
      font-size: 24px;
      font-weight: 700;
      text-align: center;
      background-color: darkgray;
      color: black;
    }
  }

  .current-score-container {
    flex-grow: 1;
    display: flex;
    align-items: center;

    .current-score {
      font-size: 16px;
      font-weight: 400;
      text-align: center;
      color: black;
      border: 1px solid black;
    }
  }

}