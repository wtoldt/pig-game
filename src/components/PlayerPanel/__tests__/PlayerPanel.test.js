import { render } from '@testing-library/react';
import PlayerPanel from '../PlayerPanel';

const defaultProps = {
  player: {
    name: 'P9',
    active: false,
    left: true,
    right: false
  }
};

function getWrapper(props={}) {
  return render(
    <PlayerPanel {...defaultProps} {...props} />
  );
}

describe('PlayerPanel Component', () => {

  describe('Snapshot testing', () => {
    it('Default snapshot', () => {
      const { container } = getWrapper();
      expect(container).toMatchSnapshot();
    });
  });
});
