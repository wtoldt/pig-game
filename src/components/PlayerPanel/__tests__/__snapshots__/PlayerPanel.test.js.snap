// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerPanel Component Snapshot testing Default snapshot 1`] = `
<div>
  <div
    class="pg-player-panel"
  >
    <div
      class="player-name-container"
    >
      <div
        class="player-name"
      >
        P9
      </div>
      <div
        class="overall-score"
      />
    </div>
    <div
      class="current-score-container"
    >
      <div
        class="current-score"
      />
    </div>
  </div>
</div>
`;
